### PR TITLE
feat(contracts): add documentation for all contracts versions

### PR DESCRIPTION
**Pull request by clemsos**
# Description

This adds a `docs` task to the `@unlock-protocol/contracts` package that generates md docs from docstring for all Unlock and PublicLock versions.

The docs will be released directly on npm in the `docs` folder of the contracts package

Also added a script to create a pull request in the docs repo with latest changes   
see https://github.com/unlock-protocol/docs/pull/31

```
sh scripts/docs-contract-pr.sh
```

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by clemsos** [comment]
need to rebase once #8806 is merged

**Comment by clemsos** [comment]
\rebase

File: scripts/docs-contracts-pr.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+git clone git@github.com:unlock-protocol/docs.git
+cd docs/developers/smart-contracts
+rm -rf contracts-api
+mkdir contracts-api
+cd contracts-api
+
+# get tarball from npm
+wget $(npm view @unlock-protocol/contracts dist.tarball)
+tar -xf unlock-protocol-contracts-**.tgz
+
```
Comment (clemsos) [inline scripts/docs-contracts-pr.sh:16]: for this to work we need to publish a version with the docs to npm first

Comment (clemsos) [reply scripts/docs-contracts-pr.sh:26]: nvm, I added a way to build it locally

**Comment by julien51** [comment]
Very cool!

**Review by julien51 — APPROVED** [review]
This is very cool!

**Pull request merged**